src/pages/User.jsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import { useAuth } from "../contexts";
+import "./pages-css/user.css";
+
+export const User = () => {
+  const { userData } = useAuth();
+  const [togglePass, setTogglePass] = useState(false);
+  const togglePassView = () => setTogglePass((x) => !x);
+
+  return (
+    <main className="user-main">
+      <div className="heading">User Profile</div>
+      <div className="profile-details-wrapper">
+        <h2 className="profile-heading">Profile Details</h2>
+        <div className="avatar avatar-l">
+          {userData.firstName.charAt(0) + userData.lastName.charAt(0)}
+        </div>
+        <div className="profile-details">
+          <ul className="list-heading">
+            <li>First Name :</li>
+            <li>Last Name :</li>
+            <li>Email :</li>
+            <li>Password :</li>
+          </ul>
+          <ul className="list-content">
+            <li>{userData.firstName}</li>
+            <li>{userData.lastName}</li>
+            <li>{userData.email}</li>
+            <li>
+              {togglePass
+                ? userData.password
+                : [...userData.password].map(() => "*")}
+              <div className="eye-btn">
```

Comment:
You can avoid using destructuring here as map return new array so you may use userData.password.map(() => "*") .

Reply:
`userData.password` is a string, not an array. 
So, by destructuring I'm spreading the string into an array.

Reply:
I'll just use `split('')` here.